src/prefect/tasks/prometheus/pushgateway.py

```diff
@@ -0,0 +1,203 @@
+from typing import Dict, List, Optional
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class _GaugeToGatewayBase(Task):
+    @abc.abstractmethod
+    def _push(self, **push_args):
+        pass
+
+    def __init__(
+        self,
+        pushgateway_url: str = None,
+        counter_name: str = None,
+        counter_description: str = None,
+        grouping_key: List[str] = None,
+        job_name: str = None,
+        **kwargs,
+    ):
+        self.pushgateway_url = pushgateway_url
+        self.counter_name = counter_name
+        self.counter_description = counter_description
+        self.job_name = job_name
+        self.grouping_key = grouping_key
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "pushgateway_url",
+        "counter_name",
+        "counter_description",
+        "job_name",
+        "grouping_key",
+    )
+    def run(
+        self,
+        values: List[float],
+        labels: List[Dict[str, str]],
+        pushgateway_url: str = None,
+        counter_name: str = None,
+        counter_description: str = None,
+        grouping_key: Optional[List[str]] = None,
+        job_name: str = None,
+    ) -> None:
+        """
+        Push a Gauge to prometheus [PushGateway](ttps://prometheus.io/docs/practices/pushing/).
+        This will unpack the values and labels to create all the a collector that will be pushed
+        to pushgateway.
+
+        Args:
+            - values (List[float]): List of the values to push
+            - labels (List[Dict[str, str]]): List of the labels to push attached to the values
+            - pushgateway_url (str, optional): Url of the prometheus pushgateway instance
+            - counter_name (str, optional): Name of the counter
+            - counter_description (str, optional): description of the counter
+            - grouping_key (str, optional): List of the key used to calculate the grouping key
+            - job_name (str, optional): Name of the job
+
+        Returns:
+            - None
+
+        Raises:
+            - ValueError: if pushgateway_url or counter_name are empty and values and labels list
+                    doesn't have the same length
+        """
+        if not pushgateway_url:
+            raise ValueError("You need to provide the pushgateway_url parameter.")
+
+        if not counter_name:
+            raise ValueError("You need to provide the counter_name parameter.")
+
+        if not counter_description:
+            counter_description = counter_name
+
+        if len(values) != len(labels):
+            raise ValueError(
+                f"Values and labels need to have the same lenght. Values lenght is {len(values)}, "
+                f"labels lenght is {len(labels)}."
+            )
+
+        if len(values) == 0 or len(labels) == 0:
+            return
+
+        grouping_key_values = (
+            {key: labels[0][key] for key in grouping_key} if grouping_key else {}
+        )
```

Comment:
Grouping keys pull from the first label? This isn't clear from the docstring. Is this mirroring prometheus behavior?

Reply:
Indeed. This is true. I updated the function to support multiple grouping keys and send all of them.